src/org/authInfo.ts

```diff
@@ -326,16 +332,25 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
 
   /**
    * Get a list of all authorizations based on auth files stored in the global directory.
+   * One can supply a filter (see @param orgAuthFilter) and calling this function without
+   * a filter will return all authorizations.
+   *
+   * @param orgAuthFilter a boolean value or a predicate function that filters authorizations
+   *        For boolean value of true, return active org authorizations, false returns inactive
+   *        org authorizations. For a predicate function as (orgAuthorization: {OrgAuthorization}): boolean return
+   *        true for those org authorizations that are to be retained.
```

Comment:
This is an extremely complex parameter that radically changes what this function returns in completely unintuitive ways. The logic around this parameter in `listAllAuthorizations()` is not easy to read or follow.

What is the reasoning behind adding it?
For W-9670267 or W-9670141 just having the `isExpired` property be returned as part of the authorization or org should be more than enough.

Reply:
How many times do we call listAllAuthorizations and then run some sot of filter across the collection.? I thought it would be useful to allow the dev to apply a filter as part of the call.

Reply:
Does not change the result returned and in practice it is not that hard to use.

Reply:
Here is how I plan on using this 
```
      const authorizations = await AuthInfo.listAllAuthorizations(
        (orgAuth) => !orgAuth.error && !orgAuth.isDevHub && !orgAuth.isExpired
      );
```

Reply:
Why not just allow the user to pass in a function to do the filtering and then just do something like
```JavaScript
if (typeof orgAuthFilter === 'function') {
  final = final.filter(orgAuthFilter);
}

return final;
```

The logic used to account for both booleans and functions being passed in makes trying to understand what exactly will be returned and when kinda convoluted.